### PR TITLE
CosmosChainProcessor Connection message handlers

### PR DESCRIPTION
**Pull request by agouin**
Similar to `channelStateCache`, maintains `connectionStateCache` which will be used by the `PathProcessor` to determine if it can proceed with channel handshakes on a given connection.

Retains connection messages so that the `PathProcessor` will be able to correlate and execute relevant connection handshakes between clients.

### CODE REVIEW

File: relayer/chains/cosmos/msg_handlers_connection.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+func (ccp *CosmosChainProcessor) handleMsgConnectionOpenInit(p msgHandlerParams) bool {
+	ci := p.messageInfo.(*connectionInfo)
+	k := ci.connectionKey()
+	// Construct the start of the MsgConnectionOpenTry for the counterparty chain.
+	// PathProcessor will determine if this is needed.
+	// For example, if a MsgConnectionOpenTry or MsgConnectionOpenConfirm is not detected on the counterparty chain, and
+	// a MsgConnectionOpenAck is not detected yet on this chain,
+	// a MsgConnectionOpenTry will be sent to the counterparty chain
+	// using this information with the connection init proof from this chain added.
```
Comment (jtieri) [inline relayer/chains/cosmos/msg_handlers_connection.go]: These comments almost read like they would be better off as Go doc top-level comments for the entire methods.

File: relayer/chains/cosmos/cosmos_chain_processor.go [hunk 2]
```diff
@@ -41,12 +44,13 @@ func NewCosmosChainProcessor(log *zap.Logger, provider *cosmos.CosmosProvider, r
 		return nil, fmt.Errorf("error getting cosmos client: %w", err)
 	}
 	return &CosmosChainProcessor{
-		log:               log.With(zap.String("chain_id", provider.ChainId())),
-		chainProvider:     provider,
-		cc:                cc,
-		pathProcessors:    pathProcessors,
-		latestClientState: make(latestClientState),
-		channelStateCache: make(processor.ChannelStateCache),
+		log:                  log.With(zap.String("chain_id", provider.ChainId())),
```
Comment (jtieri) [inline relayer/chains/cosmos/cosmos_chain_processor.go]: One thing I've been wondering is with the introduction of the chain name, as the identifier we are using for many of the CLI commands and the mapping of the chains in the config, do we want to start using the chain name for logging vs. the chain ID?

I don't think that's within the scope of this PR but something I thought was worth discussing.

**Review by jtieri — APPROVED** [review]
Doesn't look like we are introducing any new concepts or abstractions, mostly just extending the message handlers to include the Connection msgs so this LGTM.

File: relayer/chains/cosmos/cosmos_chain_processor.go (hunk 2 above)
Comment (agouin) [reply relayer/chains/cosmos/cosmos_chain_processor.go]: Would definitely be helpful, we could include both here.

File: relayer/chains/cosmos/msg_handlers_connection.go (hunk 1 above)
Comment (agouin) [reply relayer/chains/cosmos/msg_handlers_connection.go]: good call, will update

**Review by DavidNix — APPROVED** [review]

**Pull request merged**